### DISCUSSION TITLE
Is it possible to adjust the speaking speed?

### DISCUSSION THREAD

**Discussion by godspirit00** [neonbjb/tortoise-tts#95] [General]
Hello, thanks for the great work! This is the best sounding TTS I have ever heard on Github! 
I have a question.
Is it possible to adjust the speaking speed, like slowing the speech down a bit by passing a number to the api?
Thank you very much!

**Comment by jnordberg** [comment]
You can't control it directly but you can try making your own voice using samples that speak slower, it will try to mimic what you feed it including the speed and pacing.

**Comment by neonbjb** [comment]
Hey there, Johan is right - the best way to control speech speed is to use a different voice with a slower speaking rate. One thing you might try is manually slowing your reference voice down in audacity then feeding that into Tortoise. I haven't tried that but would be interested to hear what the results sound like!

**Reply by godspirit00** [reply]
Hi, I tried feeding some reference wavs of slow speech into Tortoise, and the result speech is still rather fast, faster than the reference speech I provided to it.

Here is one of the reference wavs:

https://user-images.githubusercontent.com/16176151/173033644-feae062e-ec85-4145-897a-8d191d7fa9ea.mp4

This is the result speech:

https://user-images.githubusercontent.com/16176151/173033720-85885fe0-22bf-4f44-8718-43ffc46edb15.mp4

**Reply by arkilis** [reply]
Also I am facing the same issue, have you found a solution yet? Thanks!

**Comment by neonbjb** [comment]
Huh, that's very strange. The only explanation I can offer is maybe this voice was part of the training dataset, so the model has "memorized" how it is supposed to sound and is ignoring your example in that regard. Or maybe this proposed method just doesn't work. :)

Where is this voice from? I may be able to determine whether or not this is true.

**Reply by godspirit00** [reply]
This comes from the recording of a textbook for English learners in China. I cut the audio and made it into a small dataset. The total length is around 5 hours and there are about 2800 clips. 
I'm not sure if I'm allowed to share the whole dataset... But maybe a few clips from it can do?

**Comment by neonbjb** [comment]
No need to share i was just curious. AFAIK nothing like that was in my dataset.

**Reply by godspirit00** [reply]
I tried with another voice, and the result is similar. It is faster than the reference, and it seems that it doesn't speak in the same accent as the reference.

One of the reference wavs:


https://user-images.githubusercontent.com/16176151/173239271-899a4a66-f4d7-488d-a71b-a356b2982201.mp4


The result:

https://user-images.githubusercontent.com/16176151/173239281-ae3501d2-38bb-4a75-b489-d5e58ab8b40c.mp4

I did it by generating the conditioning latent first, not sure if it is the cause?

**Comment by I-Have-No-Idea-What-IAmDoing** [comment]
Could you do some prompt engineering to slow it down like "[I am talking slowly]"?

**Reply by godspirit00** [reply]
No luck, still the same.

**Comment by neonbjb** [comment]
Oh damn, I completely forgot. There's one really important setting for speaking speed: 
https://github.com/neonbjb/tortoise-tts/blob/main/tortoise/api.py#L352

Would recommend you play around with that (in combination with the above suggestions). Sorry, I completely forgot about this.

**Reply by godspirit00** [reply]
I set the `length_penalty` to 0.2, and uses the reference wavs posted above, and it seems that the result is almost the same...
Not sure if I'm doing it right.

https://user-images.githubusercontent.com/16176151/174037192-7eccd94f-0a42-49b5-8d7e-2ad5e6af07b4.mp4

**Reply by neonbjb** [reply]
Ah, oh well. This setting definitely can produce faster speech but there must be a hard cap on how fast the model is willing to go. Sorry for wasting your time.

**Reply by godspirit00** [reply]
Thank you for your help anyway. The work is already very amazing and worth trying every possible way.

**Comment by FurkanGozukara** [comment]
@godspirit00 did you find how much training data is good? Is more better?

Also do you have a script to generate speech from subtitles such as from vtt or srt?

**Comment by manmay-nakhashi** [comment]
now you can control speaking rate :smile: :rocket:

**Reply by neonbjb** [reply]
nice job!

**Reply by godspirit00** [reply]
Great!! Thanks for the great job!

**Reply by jihwoanahn** [reply]
How do we parse the prompt for speaking rate? I couldn't find a prompt guide... :/

**Reply by aliig** [reply]
I found a commit from July implementing a speaking_rate parameter, but it's since been removed. Is this controlled with a new parameter now? @manmay-nakhashi

**Reply by seetimee** [reply]
same question

**Reply by totomtornado** [reply]
can someone explain?

**Reply by eata7** [reply]
control how?
